### PR TITLE
feat(saved-jobs): add saved/bookmarked jobs scraping with pagination and progress

### PR DESCRIPTION
**Pull request by IfThingsThenStuff**
Thanks for your work here - useful tool.  Appreciate your efforts.  I wanted the ability to read out my saved jobs - so, I added it.  It will handle multiple pages.  

Let me know if this is aligned to what you would like to include.  Let me know of any changes you think are needed.

  ## Summary
  - Add `scrape_saved_jobs` to `LinkedInExtractor` — scrapes the LinkedIn jobs tracker page, extracts job IDs from link hrefs, and paginates through results using numbered page buttons
  - Add `get_saved_jobs` MCP tool with progress reporting via `on_progress` callback
  - Cap `total_pages` with `max_pages` for accurate progress percentages
  - Use `Set` for O(1) job ID deduplication in the DOM polling function
  - Add navigation delay between page clicks consistent with other scraping methods

  ## Test plan
  - [x] `test_scrape_saved_jobs_single_page` — single page with progress callback
  - [x] `test_scrape_saved_jobs_paginates` — multi-page with progress and ID collection
  - [x] `test_scrape_saved_jobs_timeout_stops_gracefully` — timeout returns partial results
  - [x] `test_scrape_saved_jobs_stops_at_max_pages_despite_more_buttons` — respects max_pages cap
  - [x] `test_scrape_saved_jobs_empty` — empty results
  - [x] `test_get_saved_jobs` — tool-level success path
  - [x] `test_get_saved_jobs_error` — session expired error handling
  - [x] Full suite: 112/112 passing

<h3>Greptile Summary</h3>

Adds `get_saved_jobs` MCP tool to scrape saved/bookmarked jobs from LinkedIn's job tracker with pagination and progress reporting.

**Key Changes:**
- **Pagination**: Navigates through numbered page buttons, extracting job IDs from link hrefs (`/jobs/view/<id>/`)
- **Deduplication**: Uses `Set` for O(1) job ID lookups in both JavaScript extraction and Python filtering
- **Progress Reporting**: Implements `on_progress` callback with accurate page counts capped by `max_pages`
- **Error Handling**: Gracefully handles timeouts, missing buttons, and empty results
- **Navigation**: Adds 2s delay between page clicks consistent with other scraping methods

**Implementation Quality:**
- Exposes `max_pages` parameter (default 10) for user control
- Embeds job IDs in text sections for LLM visibility
- Returns both structured `job_ids` list and formatted text
- Comprehensive test suite: 6 new tests covering pagination, timeouts, edge cases
- Full test suite passing: 112/112
- Documentation updated in README.md, AGENTS.md, docs/docker-hub.md, and CLAUDE.md per development workflow

**Previous Review Items Addressed:**
- ✅ Set-based deduplication in `_EXTRACT_JOB_IDS_JS` (lines 389-390)
- ✅ Exposed `max_pages` parameter in tool signature (line 75)
- ✅ Documentation updates completed across all required files

<h3>Confidence Score: 5/5</h3>

- This PR is safe to merge with no identified issues
- Score reflects comprehensive test coverage (112/112 tests passing including 6 new tests), complete documentation updates per development workflow, robust error handling with graceful degradation, efficient O(1) deduplication using Sets, proper pagination logic with multiple safety breaks, and all previous review comments fully addressed
- No files require special attention

<h3>Important Files Changed</h3>

| Filename | Overview |
|----------|----------|
| linkedin_mcp_server/scraping/extractor.py | Added `scrape_saved_jobs` method with robust pagination logic, Set-based O(1) deduplication, proper error handling, and progress callbacks |
| linkedin_mcp_server/tools/job.py | Added `get_saved_jobs` MCP tool with exposed `max_pages` parameter, progress reporting, and consistent error handling |
| tests/test_scraping.py | Added comprehensive test suite with 5 tests covering single-page, pagination, timeout, max_pages cap, and empty results scenarios |
| tests/test_tools.py | Added tool-level tests for `get_saved_jobs` success path and error handling |

</details>

<h3>Flowchart</h3>

```mermaid
%%{init: {'theme': 'neutral'}}%%
flowchart TD
    Start([Start]) --> Navigate[Navigate to jobs-tracker]
    Navigate --> ExtractPage1[Extract page 1 text and IDs]
    ExtractPage1 --> CountButtons[Count pagination buttons]
    CountButtons --> CalcTotal[Calculate total_pages cap]
    CalcTotal --> ReportP1[Report progress page 1]
    ReportP1 --> CheckMore{More pages?}
    
    CheckMore -->|Yes| CheckButton{Button exists?}
    CheckButton -->|No| Append[Append ID summary]
    CheckButton -->|Yes| ClickButton[Click page button]
    ClickButton --> WaitDelay[Wait nav delay]
    WaitDelay --> WaitNewIDs{Wait for new IDs}
    
    WaitNewIDs -->|Timeout| Append
    WaitNewIDs -->|Success| Scroll[Scroll to bottom]
    Scroll --> ExtractText[Extract page text]
    ExtractText --> ExtractIDs[Extract job IDs]
    ExtractIDs --> FilterDups[Filter duplicates]
    FilterDups --> CheckNewIDs{New IDs?}
    
    CheckNewIDs -->|No| Append
    CheckNewIDs -->|Yes| AddIDs[Add to all_job_ids]
    AddIDs --> ReportProgress[Report progress]
    ReportProgress --> CheckMore
    
    CheckMore -->|No| Append
    Append --> BuildSections[Build sections dict]
    BuildSections --> Return([Return result])
```

<sub>Last reviewed commit: 5e68717</sub>

### CODE REVIEW

File: linkedin_mcp_server/tools/job.py [hunk 1]
```diff
@@ -64,6 +64,48 @@ async def get_job_details(job_id: str, ctx: Context) -> dict[str, Any]:
...
+                await ctx.report_progress(progress=pct, total=100, message=msg)
+
+            result = await extractor.scrape_saved_jobs(on_progress=_report)
+
+            await ctx.report_progress(progress=100, total=100, message="Complete")
+
+            return result
+
+        except Exception as e:
+            return handle_tool_error(e, "get_saved_jobs")
```
[the remark this answers is not in the document]
Comment (IfThingsThenStuff) [reply linkedin_mcp_server/tools/job.py]: Documentation added

File: linkedin_mcp_server/tools/job.py [hunk 2]
```diff
@@ -64,6 +64,48 @@ async def get_job_details(job_id: str, ctx: Context) -> dict[str, Any]:
...
 
+    @mcp.tool(
+        annotations=ToolAnnotations(
+            title="Get Saved Jobs",
+            readOnlyHint=True,
+            destructiveHint=False,
+            openWorldHint=True,
+        )
+    )
+    async def get_saved_jobs(ctx: Context) -> dict[str, Any]:
```
[the remark this answers is not in the document]
Comment (IfThingsThenStuff) [reply linkedin_mcp_server/tools/job.py]: fixed

File: linkedin_mcp_server/scraping/extractor.py [hunk 3]
```diff
@@ -384,6 +385,140 @@ async def search_jobs(
...
         }
 
+    _EXTRACT_JOB_IDS_JS = """() => {
+        const ids = [];
+        document.querySelectorAll('a[href*="/jobs/view/"]').forEach(a => {
+            const match = a.href.match(/\\/jobs\\/view\\/(\\d+)/);
+            if (match && !ids.includes(match[1])) ids.push(match[1]);
+        });
+        return ids;
+    }"""
```
[the remark this answers is not in the document]
Comment (IfThingsThenStuff) [reply linkedin_mcp_server/scraping/extractor.py:396]: fixed

**Comment by IfThingsThenStuff** [comment]
Hey there @stickerdaniel  - hope you're doing well.  Is there anything I can do to help get this merged in sir?  Thanks in advance.  Let me know.
